src/schemas.jl

```diff
@@ -440,18 +462,38 @@ function _generate_schema_version_definitions(schema_version::SchemaVersion, par
 end
 
 function _generate_validation_definitions(schema_version::SchemaVersion)
-    field_violation_check_statements = Expr[]
-    for (fname, ftype) in pairs(required_fields(schema_version))
-        fname = Base.Meta.quot(fname)
-        push!(field_violation_check_statements, quote
-            S = $Legolas.accepted_field_type(sv, $ftype)
-            result = $Legolas._check_for_expected_field(ts, $fname, S)
-            isnothing(result) || return $fname => result
+    # When `fail_fast == true`, return first violation found rather than all violations
+    _violation_check = (; fail_fast::Bool) -> begin
+        violations = Expr[]
+        myvector = gensym()
+        push!(violations, :($myvector = Any[]))
+        for (fname, ftype) in pairs(required_fields(schema_version))
+            fname = Base.Meta.quot(fname)
+            push!(violations, quote
+                S = $Legolas.accepted_field_type(sv, $ftype)
+                result = $Legolas._check_for_expected_field(ts, $fname, S)
+                if !isnothing(result) && $(fail_fast)
+                    return $fname => result
+                elseif !isnothing(result)
+                    push!($myvector, ($fname => result))
+                end
+            end)
+        end
+        push!(violations, quote
+            return $(fail_fast) ? nothing : $myvector
         end)
+        return violations
     end
+
     return quote
         function $(Legolas).find_violation(ts::$(Tables).Schema, sv::$(Base.Meta.quot(typeof(schema_version))))
-            $(field_violation_check_statements...)
+            $(_violation_check(; fail_fast=true)...)
+            return nothing
+        end
+
+        # Multiple violation reporting
+        function $(Legolas).find_violations(ts::$(Tables).Schema, sv::$(Base.Meta.quot(typeof(schema_version))))
+            $(_violation_check(; fail_fast=false)...)
             return nothing
         end
```

Comment:
IMO we should use the code from `_violation_check` and generate a single function `_find_violations` which is then called by:
```julia
find_violation(ts::Tables.Schema, sv::SchemaVersion) = _find_violations(ts, vs; fail_fast=true)
find_violations(ts::Tables.Schema, sv::SchemaVersion) = _find_violations(ts, vs; fail_fast=false)
```

Reply:
Here's a first pass at the `_violation_check` function:
```julia
SV = Base.Meta.quot(typeof(schema_version))
quote
    function _violation_check(ts::$(Tables).Schema, sv::$SV; fail_fast::Bool)
        violations = Pair{Symbol,Union{Type,Missing}}[]
        for (fname, ftype) in pairs(required_fields(sv))
            S = Legolas.accepted_field_type(sv, ftype)
            result = Legolas._check_for_expected_field(ts, fname, S)
            if !isnothing(result)
                push!(violations, fname => result)
                fail_fast && break
            end
        end
        return violations
    end
end
```

Reply:
so there actually is a difference in what gets returned by these two functions---find_violation returns `nothing` or the first violation, where `find_violations` returns the whole (potentially empty) list of violations. Since we're planning to deprecate `find_violation` anyway, I'd like to just use what's here already and then remove the extraneous lines when when we remove find_violations outright. (I attempted to :just: pull this chunk in, but it didn't work out of the box, and I don't think the priority is high enough to spend more time fighting it.)

Reply:
@hannahilea I was intending for you to the above in conjunction with these:
```julia
function Legolas.find_violation(ts::Tables.Schema, sv::SchemaVersion)
    violations = _violation_check(sv; fail_fast=true)
    return isempty(violations) ? nothing : first(violations)
end

function Legolas.find_violations(ts::Tables.Schema, sv::SchemaVersion)
    return _violation_check(sv; fail_fast=false)
end
```
I do think this is important to make this change to help transition this package into a more maintainable state by others.

Reply:
I agree that this would be nice, but :just: copying in the changes here did not successfully run for me; I think there's probably something basic that I'm neglecting to also add or something. A little bit of playing around didn't get me there, and I don't have any more time to allocate to this side-quest 😅 .  Since 

> to help transition this package into a more maintainable state by others

is not the current goal, and I don't think the changes are _less_ readable than before, I'd like to merge it as is and do any readability changes in future PRs.

Reply:
Unrelated note for future readers: my above statement that 
> so there actually is a difference in what gets returned by these two functions---find_violation returns nothing or the first violation, where find_violations returns the whole (potentially empty) list of violations

is now incorrect---I updated it so that `find_violations` _also_ returns `nothing` when no violations are found, rather than an empty list. That's more in line with our other validation-type functions, and will make deprecation fixing easier for anyone who depends on this.

Reply:
merged in @omus's suggestions via #87